The enforcer rule integration tests in "enforcer-rules/src/it" fails in Java 11

# bom-project-using-spring-repository/pom.xml

```
[INFO]   The post-build script did not succeed. assert buildLog.text.contains("Linkage Checker rule found 850 errors:")
...
       |        [ERROR] Linkage Checker rule found 846 errors:
       |        Class org.openjdk.jcstress.infra.results.I_Result is not found;
...
```

How to see the difference:

```
# 
$ mvn clean install -Dmaven.test.skip -Dinvoker.skip    
...
# modify the version to the locally installed enforcer rule
$suztomo-macbookpro44% git diff  
diff --git a/enforcer-rules/src/it/bom-project-using-spring-repository/pom.xml b/enforcer-rules/src/it/bom-project-using-spring-repository/pom.xml
index a734c6e5..84ebb516 100644
--- a/enforcer-rules/src/it/bom-project-using-spring-repository/pom.xml
+++ b/enforcer-rules/src/it/bom-project-using-spring-repository/pom.xml
@@ -72,7 +72,7 @@
           <dependency>
             <groupId>com.google.cloud.tools</groupId>
             <artifactId>linkage-checker-enforcer-rules</artifactId>
-            <version>@project.version@</version>
+            <version>1.5.5-SNAPSHOT</version>
           </dependency>
         </dependencies>
         <executions>
$ mvn verify -Pspring > /tmp/errors_java11
```

Compared the output between Java 8 and Java 11: https://gist.github.com/suztomo/f23c1a5b67b54e7b6dc68bbfe83145c1.

It's `java.util.concurrent.Flow$Publisher`, `java.util.concurrent.Flow$Processor`, `java.util.concurrent.Flow$Subscription`, and `java.util.concurrent.Flow$Subscriber` classes that were not found in Java 8. [JEP 266: More Concurrency Updates, implemented in Java 9](http://openjdk.java.net/jeps/266) explains the classes.

## The difference in the dependency graphs

Java runtime version itself does not change the dependency graph version but the dependency graphs differ between Java 8 and Java 11 runtime.

```
<     The valid symbol is in org.jboss.marshalling:jboss-marshalling:jar:1.4.11.Final at io.projectreactor.netty:reactor-netty-core:1.0.0-M2 (compile) / io.netty:netty-handler:4.1.51.Final (compile) / io.netty:netty-codec:4.1.51.Final (compile) / org.jboss.marshalling:jboss-marshalling:1.4.11.Final (compile, optional) but it was not selected because the path contains an optional dependency
---
>     The valid symbol is in org.jboss.marshalling:jboss-marshalling:jar:2.0.5.Final at io.projectreactor.netty:reactor-netty-core:1.0.0-M2 (compile) / io.netty:netty-handler:4.1.51.Final (compile) / io.netty:netty-codec:4.1.51.Final (compile) / org.jboss.marshalling:jboss-marshalling:2.0.5.Final (compile, optional) but it was not selected because the path contains an optional dependency
```

It turned out that netty-parent has profile to update certain property to specify  the version:

```
    <profile>
      <id>java11</id>
      <activation>
        <jdk>11</jdk>
      </activation>
      <properties>
        <argLine.java9.extras />
        <!-- Export some stuff which is used during our tests -->
        <argLine.java9>--illegal-access=deny ${argLine.java9.extras}</argLine.java9>
        <!-- Not use alpn agent as Java11+ supports alpn out of the box -->
        <argLine.alpnAgent />
        <forbiddenapis.skip>true</forbiddenapis.skip>
        <!-- Needed because of https://issues.apache.org/jira/browse/MENFORCER-275 -->
        <enforcer.plugin.version>3.0.0-M3</enforcer.plugin.version>
        <!-- 1.4.x does not work in Java10+ -->
        <jboss.marshalling.version>2.0.5.Final</jboss.marshalling.version>
        <!-- pax-exam does not work on latest Java11 build -->
        <skipOsgiTestsuite>true</skipOsgiTestsuite>
      </properties>
    </profile>
```

# return-type-mismatch/pom.xml

```
[INFO] Building: return-type-mismatch/pom.xml
[INFO] run post-build script verify.groovy
[INFO]   The post-build script did not succeed. assert buildLog.contains('''\ java.nio.ByteBuffer's method position(int) is expected to return java.nio.ByteBuffer\ but instead returns java.nio.Buffer; referenced by 8 class files com.google.protobuf.AllocatedBuffer (com.google.protobuf:protobuf-java:3.12.4) com.google.protobuf.BinaryWriter (com.google.protobuf:protobuf-java:3.12.4) com.google.protobuf.ByteBufferWriter (com.google.protobuf:protobuf-java:3.12.4) com.google.protobuf.CodedInputStream (com.google.protobuf:protobuf-java:3.12.4) com.google.protobuf.CodedOutputStream (com.google.protobuf:protobuf-java:3.12.4) com.google.protobuf.IterableByteBufferInputStream (com.google.protobuf:protobuf-java:3.12.4) com.google.protobuf.NioByteString (com.google.protobuf:protobuf-java:3.12.4) com.google.protobuf.Utf8 (com.google.protobuf:protobuf-java:3.12.4)''')
       |        |

```

This protobuf-java problem only occurs in Java 8.

Any way to limit the execution only in Java 8? (JUnit has Assume) ->[`invoker.java.version`](https://maven.apache.org/plugins/maven-invoker-plugin/run-mojo.html) does this.


How to reproduce the failure

```
suztomo-macbookpro44% pwd                                                              
/Users/suztomo/cloud-opensource-java/enforcer-rules
suztomo-macbookpro44% which java                                                       
/Library/Java/JavaVirtualMachines/jdk-11-latest/Contents/Home/bin/java
```

--- a/enforcer-rules/src/it/bom-project-using-spring-repository/verify.groovy
+++ b/enforcer-rules/src/it/bom-project-using-spring-repository/verify.groovy
@@ -12,4 +12,8 @@ assert !buildLog.text.contains(
 
 assert !buildLog.text.contains("NullPointerException")
 
-assert buildLog.text.contains("Linkage Checker rule found 850 errors:")
+// 4 linkage errors are references to java.util.concurrent.Flow class, which does not exist in
+// Java 8 runtime yet.
+def expectedErrorCount = System.getProperty("java.version").startsWith("1.8.") ? 850 : 846
+
+assert buildLog.text.contains("Linkage Checker rule found $expectedErrorCount errors:")

--- a/enforcer-rules/src/it/return-type-mismatch/invoker.properties
+++ b/enforcer-rules/src/it/return-type-mismatch/invoker.properties
@@ -1,2 +1,3 @@
 invoker.goals=verify
 invoker.buildResult=failure
+invoker.java.version=1.8

